src/app/hooks.ts

```diff
@@ -7,7 +7,12 @@
 import { useDispatch, useSelector } from "react-redux"
 
 import type { AppDispatch, RootState } from "./store"
+import settingsSlice, { type SettingsState } from "./slices/settings"
 
 // Use throughout your app instead of plain `useDispatch` and `useSelector`
 export const useAppDispatch = useDispatch.withTypes<AppDispatch>()
 export const useAppSelector = useSelector.withTypes<RootState>()
+
+export function useSettings(): SettingsState {
+  return useSelector(settingsSlice.selectors.selectWhole)
+}
```

Comment:
1. the type is inferred
1. use the exported selector
```suggestion
import { selectSettings } from "./slices/settings"

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = useDispatch.withTypes<AppDispatch>()
export const useAppSelector = useSelector.withTypes<RootState>()

export function useSettings() {
  return useSelector(selectSettings)
}
```